src/java/org/apache/cassandra/config/GuardrailsOptions.java

```diff
@@ -653,6 +691,25 @@ private static void validateWarnLowerThanFail(long warn, long fail, String name)
                                                       "than the fail threshold %d", warn, name, fail));
     }
 
+    private static void validateWarnGreaterThanFail(long warn, long fail, String name)
+    {
+        if (warn == -1|| fail == -1)
+            return;
+
+        if (fail > warn)
+            throw new IllegalArgumentException(format("The warn threshold %d for %s_warn_threshold should be greater " +
+                                                      "than the fail threshold %d", warn, name, fail));
+    }
+
+    private static void validateMinRFVersesDefaultRF(int fail, String name)
+    {
+        if (fail > DatabaseDescriptor.getDefaultKeyspaceRF())
+        {
+            throw new ConfigurationException(String.format("%s_fail_threshold to be set (%d) cannot be greater than default_keyspace_rf (%d)",
```

Comment:
We should throw `IllegalArgumentException` or some other generic Java exception, so `ConfigurationException` doesn't leak to JMX. At startup, [`DatabaseDescriptor.applyGuardrails`](https://github.com/apache/cassandra/blob/0965710345b1e9887d6bbd53b3df45278f8743f4/src/java/org/apache/cassandra/config/DatabaseDescriptor.java#L931-L938) would catch that `IllegalArgumentException` and throw a `ConfigurationException` instead.